valhalla/mjolnir/landmark_database_builder.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <boost/property_tree/ptree.hpp>
+#include <cstdint>
+#include <iostream>
+#include <list>
+#include <string>
+#include <vector>
+
+#include "baldr/graphconstants.h"
+#include "filesystem.h"
+#include "mjolnir/util.h"
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/geometries.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include <geos_c.h>
+
+namespace valhalla {
+namespace mjolnir {
+enum class AccessMode { ReadWriteCreate, ReadOnly, ReadWrite };
+
+struct Landmark {
+  std::string name;
+  std::string type;
```

Comment:
in the next pr where you add parsing real data this will need to be come an enum of type uint8_t